src/dataflow/src/sink/kafka.rs

```diff
@@ -856,9 +1129,50 @@ where
                     accum.meet(&history.durability_frontier())
                 });
 
+            // Can't use `?` when the return type is a `bool` so use a custom try operator
+            macro_rules! bail_err {
+                ($expr:expr) => {
+                    match $expr {
+                        Ok(val) => val,
+                        Err(_) => {
+                            s.activator.activate();
+                            return true;
+                        }
+                    }
+                };
+            }
+
+            if is_active_worker {
+                if let KafkaSinkStateEnum::Init(ref init) = s.sink_state {
+                    if s.transactional {
+                        bail_err!(s.retry_on_txn_error(|p| p.init_transactions()).await);
+                    }
+
+                    let latest_ts = match s.determine_latest_consistency_record().await {
+                        Ok(ts) => ts,
+                        Err(e) => {
+                            s.shutdown_flag.store(true, Ordering::SeqCst);
+                            info!("shutting down kafka sink while initializing: {}", e);
+                            return true;
+                        }
+                    };
+                    let consistency_state = init
+                        .as_ref()
+                        .cloned()
```

Comment:
nit: `.as_ref().cloned()` is the same as just `.clone()`

Reply:
duh yes thanks